feg/gateway/services/n7_n40_proxy/n7/model_conversion.go

```diff
@@ -0,0 +1,595 @@
+				ActivationTime:   activationTime,
+				DeactivationTime: deactivationTime,
+			}
+			staticRules = append(staticRules, staticRule)
+			continue
+		}
+
+		dynamicRule := &protos.DynamicRuleInstall{
+			PolicyRule:       policyRule,
+			ActivationTime:   activationTime,
+			DeactivationTime: deactivationTime,
+		}
+		dynamicRules = append(dynamicRules, dynamicRule)
+	}
+
+	return staticRules, dynamicRules, rulesToRemove
+}
+
+func GetActivateDeactivationTime(
+	policyDecision *n7_sbi.SmPolicyDecision,
+	pccRule *n7_sbi.PccRule,
+) (activationTime, deactivationTime *timestamp.Timestamp) {
+	if pccRule.RefCondData == nil || *pccRule.RefCondData == "" {
+		return
+	}
+	condData, found := policyDecision.Conds.Get(*pccRule.RefCondData)
+	if !found {
+		return
+	}
+	activationTime = ConvertToProtoTimeStamp(condData.ActivationTime)
+	deactivationTime = ConvertToProtoTimeStamp(condData.DeactivationTime)
+	return
+}
+
+func GetPolicyRuleProto(
+	policyDecision *n7_sbi.SmPolicyDecision,
+	pccRule *n7_sbi.PccRule,
+) *protos.PolicyRule {
+	// For all Ref*Data Maximum one entry is expected.
+	// These are declared as arrays for future compatibility
+	policyRule := &protos.PolicyRule{
+		Id:            pccRule.PccRuleId,
+		Priority:      getSbiUinteger(pccRule.Precedence),
+		MonitoringKey: getMonKeyFromPccRule(policyDecision, pccRule),
+		Redirect:      getTcDataFromPccRule(policyDecision, pccRule),
+		Qos:           getQosDataFromPccRule(policyDecision, pccRule),
+		FlowList:      getFlowListFromPccRule(pccRule),
+	}
+	insertChargingDataToPolicyRule(policyDecision, pccRule, policyRule)
+	policyRule.TrackingType = getTrackingTypeProto(len(policyRule.MonitoringKey) != 0, policyRule.RatingGroup != 0)
+
+	if len(policyRule.MonitoringKey) == 0 && policyRule.Redirect == nil &&
+		policyRule.Qos == nil && len(policyRule.FlowList) == 0 && policyRule.RatingGroup == 0 {
+		// must be a static rule if these are not present
+		return nil
+	}
+	return policyRule
+}
+
+func insertChargingDataToPolicyRule(
+	policyDecision *n7_sbi.SmPolicyDecision,
+	pccRule *n7_sbi.PccRule,
+	policyRule *protos.PolicyRule,
+) {
+	if pccRule.RefChgData == nil || len(*pccRule.RefChgData) == 0 || policyDecision.ChgDecs == nil {
+		return
+	}
+	chgData, found := policyDecision.ChgDecs.Get((*pccRule.RefChgData)[0])
+	if !found {
+		return
+	}
+	policyRule.RatingGroup = uint32(*chgData.RatingGroup)
+	policyRule.ServiceIdentifier = &protos.ServiceIdentifier{
+		Value: uint32(*chgData.ServiceId),
+	}
+	policyRule.Offline = getSbiBool(chgData.Offline)
+	policyRule.Online = getSbiBool(chgData.Online)
+}
+
+func getMonKeyFromPccRule(
+	policyDecision *n7_sbi.SmPolicyDecision,
+	pccRule *n7_sbi.PccRule,
+) []byte {
+	if pccRule.RefUmData != nil && len(*pccRule.RefUmData) > 0 && policyDecision.UmDecs != nil {
+		umData, found := policyDecision.UmDecs.Get((*pccRule.RefUmData)[0])
+		if found {
+			return []byte(umData.UmId)
+		}
+	}
+	return []byte{}
+}
+
+func getTcDataFromPccRule(
```

Comment:
`getTcDataFromPccRule` this name is not very descriptive . Can you add a comment on what TC is 
Something like:
```
// getTcDataFromPccRule extracts Redirect Information from the SmPolicyDecision and returns a proto SmPolicyDecision

```